### PR TITLE
Nacelle Model

### PR DESCRIPTION
**Pull request by tonyinme**
A nacelle model is included in the actuator line model.
Documentation has been added in both theory and user manuals.
Three new inputs are added to the input file: nacelle drag coefficient, nacelle area, air density.
The epsilon value is computed based on these inputs.
The forces are computed within OpenFAST.

The reg test for the NREL 5MW has been modified to include the nacelle model.
The previous test without the nacelle model also passes (nacelle model is deactivated if no inputs are found).

**This pull request needs to be approved before being able to merge:** (https://github.com/OpenFAST/openfast/pull/106)

### CODE REVIEW

File: src/ActuatorLineFAST.C [hunk 1]
```diff
@@ -874,14 +880,59 @@ ActuatorLineFAST::create_actuator_line_point_info_map() {
...
+
+              for (int j=0; j<nDim; j++) {
+
+
+                  // Compute epsilon only if drag coefficient is greater than zero
+                  if (nac_cd>0){
+                      // This model is used to set the momentum thickness
+                      // of the wake (Martinez-Tossas PhD Thesis 2017)
+                      float tmpEps = std::sqrt(2.0 / pi * nac_cd * nac_area); 
+                      std::cout<<"Epsilon nacelle "<< tmpEps<<std::endl;
```
Comment (sayerhs) [inline src/ActuatorLineFAST.C]: Remove print statement

File: src/ActuatorLineFAST.C [hunk 2]
```diff
@@ -874,14 +880,59 @@ ActuatorLineFAST::create_actuator_line_point_info_map() {
...
+              }
+              break;
+          }
+          case fast::BLADE:
+              epsilon = actuatorLineInfo->epsilon_;
+              break;
+          case fast::TOWER:
+              epsilon = actuatorLineInfo->epsilon_;
+              break;
+          case fast::ActuatorNodeType_END:
```
Comment (sayerhs) [inline src/ActuatorLineFAST.C]: This should never happen correct? And it is being checked for before?

File: src/ActuatorLineFAST.C [hunk 3]
```diff
@@ -874,14 +880,59 @@ ActuatorLineFAST::create_actuator_line_point_info_map() {
...
+                  else{
+                      epsilon = actuatorLineInfo->epsilon_;
+                      //~ epsilon.y_ = actuatorLineInfo->epsilon_;
+                      //~ epsilon.z_ = actuatorLineInfo->epsilon_;
+                      }
+
+              }
+              break;
+          }
+          case fast::BLADE:
```
Comment (sayerhs) [inline src/ActuatorLineFAST.C]: Can collapse this to 

```
case fast::BLADE:
case fast::TOWER:
     epsilon = actuatorLineInfo->epsilon_;
     break;
```

File: docs/source/user/nalu_run/nalu_inp.rst [hunk 4]
```diff
@@ -999,141 +999,7 @@ Time-step Control Options
...
-   The checkpoint file for this turbine when restarting a simulation
-   
-.. inpfile:: actuator.FAST_input_filename
-
-   The FAST input file for this turbine
-  
-.. inpfile:: actuator.turb_id
-
-   A unique turbine id for each turbine
+.. include:: ./turbine_modeling.rst
```
Comment (sayerhs) [inline docs/source/user/nalu_run/nalu_inp.rst]: A large section was removed from the docs, but the equivalent file `turbine_modeling.rst` was not committed to the repository.

Comment (tonyinme) [reply docs/source/user/nalu_run/nalu_inp.rst]: This file has now been added to the repository.

File: src/ActuatorLineFAST.C (hunk 2 above)
Comment (tonyinme) [reply src/ActuatorLineFAST.C]: Yes, this has been changed to a default statement.

File: src/ActuatorLineFAST.C (hunk 3 above)
Comment (tonyinme) [reply src/ActuatorLineFAST.C]: We are leaving this as is for readability and in the future they might differ.

File: src/ActuatorLineFAST.C (hunk 1 above)
Comment (tonyinme) [reply src/ActuatorLineFAST.C]: This has been removed.

**Comment by tonyinme** [comment]
These changes have been pushed. It is now good to go.

**Comment by tonyinme** [comment]
Hey @sayerhs, is there something else needed to approve this pull request?
The gold standard file changed for the test of the NREL 5MW.
Is there anything else that needs to be added?

**Comment by sayerhs** [comment]
@tonyinme The code looks good, but you will need to regenerate the gold file because of recent changes to `TpetraLinearSystem`.

**Comment by tonyinme** [comment]
Hey @sayerhs, just updated that.
Let me know if this works.

**Comment by jrood-nrel** [comment]
@tonyinme I'm closing this and I will do the merge in Nalu-Wind and update the gold files.

**Pull request closed without merging**